packages/did/src/Did.utils.ts

```diff
@@ -263,8 +264,53 @@ export function isVerificationKey(
 export function isEncryptionKey(
   key: Partial<NewDidKey | DidKey> & Pick<NewDidKey | DidKey, 'type'>
 ): key is NewDidEncryptionKey | DidEncryptionKey {
-  const keyType = key.type.toLowerCase()
-  return encryptionKeyTypes.some((kt) => kt === keyType)
+  return encryptionKeyTypes.some((kt) => kt === key.type)
+}
+
+export const verificationKeyTypeToChainKeyType: Record<
+  VerificationKeyType,
+  DidDidDetailsDidVerificationKey['type']
+> = {
+  sr25519: 'Sr25519',
+  ed25519: 'Ed25519',
+  ecdsa: 'Ecdsa',
+}
+
+export const chainVerificationKeyTypeToKeyType = Object.entries(
+  verificationKeyTypeToChainKeyType
+).reduce((obj, [key, value]) => ({ ...obj, [value]: key }), {}) as Record<
+  DidDidDetailsDidVerificationKey['type'],
+  VerificationKeyType
+>
+
+export type EncodedVerificationKey =
+  | { Sr25519: Uint8Array }
+  | { Ed25519: Uint8Array }
+  | { Ecdsa: Uint8Array }
```

Comment:
In my opinion this is much more preferable to the unholy
```ts
K extends any
  ? { [P in K]: V }
  : never

Reply:
Personally, being more familiar with Typescript than with writing JS directly, I like the TypedValue utility type a lot (note that something like Record is also just an example of such a utility type) and IMO it's clear what is happening from its documentation. While I had a small gripe with the repeated type forcing, I did like the fact that we solved what is a pure type system issue on the level of the type system - leaving no trace in the final JS code.

Reply:
Neither solution leaves a trace in the final JS code, I think, since all the types are removed. I see the value in fixing the type system issue using the type system itself, however I dislike the much larger downside of the solution being extremely obscure. The "RTFM" is pretty much never a good argument :)